nixos/modules/services/continuous-integration/github-runner/query_registration_token.py

```diff
@@ -0,0 +1,85 @@
+import sys, json
+import logging
+from urllib.request import urlopen, Request
+from urllib.parse import urlparse
+from urllib.error import HTTPError
+
+
+def new_registration_token_from_pat(url: str, pat: str) -> str:
+    """Queries a new registration token for a github.com repository or organisation"""
+    components = urlparse(url)
+    if components.netloc not in {"github.com", "github.com:443"}:
+        raise ValueError("Only PATs for github.com are supported!")
+
+    path_parts = [x for x in components.path.split("/") if x]
+    if len(path_parts) == 1:
+        logging.debug("Given URL is for a Github organization")
+        endpoint = (
+            "https://api.github.com/orgs/{}/actions/runners/registration-token".format(
+                path_parts[0]
+            )
+        )
+    else:
+        logging.debug("Given URL is for a single Github repository")
+        endpoint = "https://api.github.com/repos/{}/{}/actions/runners/registration-token".format(
+            *path_parts
+        )
+
+    req = Request(
+        endpoint,
+        method="POST",
+        headers={
+            "Authorization": f"token {pat}",
+            "Accept": "application/vnd.github+json",
+        },
+    )
+    logging.debug(f"Performing API call to {endpoint}")
+    try:
+        with urlopen(req) as resp:
+            return json.load(resp)["token"]
+    except HTTPError as err:
+        raise RuntimeError(
+            "Caught unexpected status {err.status} while getting registration token:\n{err.read().decode()}"
+        )
```

Comment:
```suggestion
    except HTTPError as err:
        raise RuntimeError(
            "Caught unexpected status {err.status} while getting registration token:\n{err.read().decode()}"
        ) from err
```

This is just a cosmetic change for the exception traceback, it makes it clear that the `RuntimeError` exception occurred as the result of the `HTTPError`, instead of the default which (incorrectly) indicates that the `RuntimeError` occured while trying to fix the `HTTPError`.